server/src/coqtop/CoqTop8.ts

```diff
@@ -250,17 +250,16 @@ export class CoqTop extends IdeSlave8 implements coqtop.CoqTop {
         ...this.settings.args
         ];
     }
-    if (this.settings.useDune) {
-      var binary = this.settings.dunePath
-      if (scriptPath === undefined)
-        throw new CoqtopSpawnError("", "File was not saved to local file system");
-      var args = ["coq", "top", "--toplevel=" + coqtopModule,
-        scriptPath, "--", ...coqArgs];
-    } else {
-      var binary = coqtopModule
-      var args = coqArgs
+
+    if (this.settings.useDune && scriptPath === undefined) {
+      throw new CoqtopSpawnError("", "File was not saved to local file system");
     }
 
+    const [binary, args] = this.settings.useDune ?
+      [this.settings.dunePath, ["coq", "top", "--toplevel=" + coqtopModule,
+        scriptPath, "--", ...coqArgs]] :
+      [coqtopModule, coqArgs]
+
```

Comment:
@huynhtrankhanh Since the code relied on `var`'s scoping, this seems the smallest possible change.

Had to lift error checking since TS seems to lack statement expressions, but I guess that's not too bad. I guess `function () { ... } ()` lets you encode them, but I wouldn't know if that's idiomatic.

Reply:
To be clear, if anything seems silly it's because I'm not really a TypeScript dev and I was following existing idioms.